cfme/networks/security_group.py

```diff
@@ -54,7 +54,16 @@ def all(self):
         else:
             view = navigate_to(self, 'All')
         list_networks_obj = view.entities.get_all(surf_pages=True)
-        return [self.instantiate(name=s.name) for s in list_networks_obj]
+
+        collection = (self.instantiate(name=s.name) for s in list_networks_obj)
+        return [a for a in collection if apply_filters(a, self.filters)]
```

Comment:
Required:
I am suspect It will not return security group objects when I will call without a filter.

Reply:
You are not right:
```
In [4]: def apply_filters(item, filters):
   ...:     for attr_name, val in filters.items():
   ...:         if getattr(item, attr_name) != val:
   ...:             return False
   ...:     return True
   ...: 

In [5]: class A(object):
   ...:     def __init__(self, foo):
   ...:         self.foo = foo
   ...:         

In [7]: collection = [A(str(i)) for i in range(10)]

In [13]: [a.foo for a in collection if apply_filters(a, {})]
Out[13]: ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']

In [14]: [a.foo for a in collection if apply_filters(a, {'foo': '3'})]
Out[14]: ['3']
```